app/models/concerns/profile_tailoring.rb

```diff
@@ -58,4 +58,8 @@ def update_os_minor_version(version)
 
     update!(os_minor_version: version)
   end
+
+  def value_overrides_by_ref_id
+    value_overrides.transform_keys { |v| ValueDefinition.find(v).ref_id }
```

Comment:
This will generate a query for each value override for a profile.. Not sure if caching can be used somehow?

Reply:
This is bad :see_no_evil:

Reply:
This should do a single query
```
ValueDefinition.where(id: value_overrides.keys).each_with_object
```